### ISSUE TITLE
[Transfer to DST] CAIA A11y Support 508-defect-3: Navigation menus need accessible name for DST & Sitewide

### ISSUE THREAD

**Issue by jaredcunha** [department-of-veterans-affairs/va.gov-team#59760]
## Transfer Ticket
`Needs reassigned to the new DST specialist`

### Point of contact

jaredcunha

### CAIA Accessibility Specialist
@SarahKay8 

### Severity level

3, Moderate. Should be fixed 1-3 sprints post-launch

### Details

There are often multiple `navigation` landmarks on a page, for example, when page has left-hand navigation. Using the landmarks rotor, it is difficult to distinguish the different navigation menus from each other.

<img width="1091" alt="Screenshot 2023-06-06 at 9 27 52 AM" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/1094951/e317ffd4-2c1f-433b-916e-57c5a00b82b6">

### Reproduction steps

1. Visit a page with left navigation, for example: https://www.va.gov/careers-employment/vocational-rehabilitation/
2. Turn on VoiceOver
3. Open the rotor and use the arrow keys to access the Landmarks menu
4. Note the two unnamed "navigation" options

### For the mega menu:
1. Add `aria-label="Main"` (or "Primary" or what the content team decides) to `<div id="vetnav" role="navigation">` in the `MegaMenu` component so that it is `<div id="vetnav" role="navigation" aria-label="Main">`;  or
2. Use this alternative that might better support translation plugins on a screen reader, which still provides a name to the landmark but hides the content from the virtual cursor:
```jsx
<div id="vetnav" role="navigation" aria-labelledby="vetnav-name">
   <span id="vetnav-name" aria-hidden="true">Main</span>
```

### For the left navigation menus
```jsx
<nav aria-labelledby="va-sidebarnav-header" data-template="navigation/sidebar_nav" id="va-detailpage-sidebar" data-drupal-sidebar="true" class="va-drupal-sidebarnav usa-width-one-fourth va-sidebarnav" >
...
    <h4 id="va-sidebarnav-header">Careers and employment</h4>
```
- Add `aria-labelledby` to the `<nav>` element
- Add `id` to the header

There might be some additional work required to expose this navigation landmark as a navigation landmark in the mobile context. Right now, the `<nav>` is hidden and is only shown when the user clicks the button named "In this section" which is under the Form Controls rotor, which could make it easier to miss.

### References, articles, or WCAG support

1. [WCAG SC 1.3.6 Identify Purpose Level AAA](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=131%2C412#identify-purpose)

### Type of issue

- [X] Screenreader
- [ ] Keyboard
- [ ] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by jaredcunha** [comment]
There's an old version of this ticket. Linking:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/6368

The breadcrumbs suggestion in that ticket has since been addressed.

**Comment by SarahKay8** [comment]
Hey @jaredcunha! Just assigned this to myself. Am I just validating this bug? Wanted to clarify what next steps were in terms of this ticket

**Comment by sara-amanda** [comment]
@SarahKay8 Moving to `validate` status on Sitewide CAIA pipeline, so we can transfer to @rsmithadhoc
cc: @coforma-terry

**Comment by micahchiang** [comment]
@humancompanion-usds @caw310 - This should be handed off to Site Wide. Neither the mega menu nor the side nav are part of the Design System.

**Comment by humancompanion-usds** [comment]
@micahchiang - That's fine by me. @caw310 - Can you work with the sitewide PM to reassign? Thanks.

**Comment by caw310** [comment]
@jilladams @coforma-terry , DST thinks this is actually a Site Wide issue. Am reassigning to you both.

**Comment by jilladams** [comment]
@laflannery @davidconlon FYI. I think the Left nav has always been a little blurry re: ownership -- Dave, any insights welcome.

**Comment by laflannery** [comment]
@jilladams Because this is not a component in the DS yet, we this has already been resolved on the PW side in [this ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14870), you can see it completed on [this page](https://www.va.gov/careers-employment/vocational-rehabilitation/) and in this screenshot:

![image](https://github.com/department-of-veterans-affairs/va.gov-team/assets/106678594/adb6430b-67c8-4a5a-955a-7c58c7cf3760)

Regarding the DST piece, I did open [a bug](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1999) with them to to make the same update

I believe with that information that would mean this falls on the DST until a component is made and at that point we would then update our templates to use that component.

**Comment by jilladams** [comment]
Thanks Laura. @caw310 should we reassign based on that note?

**Comment by caw310** [comment]
@jilladams it seems kind of out of order since we don't have that component yet.

**Comment by jilladams** [comment]
Ok, talked to Laura Flannery. 
* PW has mitigated this in the pages where we are able. 
* The DST will ship a component for left nav eventually, we believe.
* When they do, PW can prioritize / adopt that.

So there's nothing actionable to do on this ticket today, and I'd like to advocate we close it and move to adopt the component once it exists. Concerns?

**Comment by caw310** [comment]
Thanks @jilladams ! I would agree that we close this and open a new ticket at the appropriate time if/when needed.
